### PR TITLE
driver/at30tse75x: port to ztimer_usec

### PR DESCRIPTION
**Pull request by kfessel**
### Contribution description

port at30tse75x-driver to ztimer_usec

### Testing procedure

build and run tests/at30tse75x-driver

### Issues/PRs references

#17111

### CODE REVIEW

**Comment by aabadie** [comment]
If you want to test on hardware, this sensor is available on the samr21-xpro boards provided by IoT-LAB (Saclay site only), see https://www.iot-lab.info/docs/boards/microchip-samr21/

File: tests/test_interactive.config [hunk 1]
```diff
@@ -1 +1 @@
-CONFIG_MODULE_TEST_UTILS_INTERACTIVE_SYNC=y
+CONFIG_MODULE_TEST_UTILS_INTERACTIVE_SYNC=N
```
Comment (fjmolinas) [inline tests/test_interactive.config:1]: Not sure if it got in my mistake or you wanted to do it only for the test application for this driver?

Comment (kfessel) [reply tests/test_interactive.config:1]: I wanted to test it on iot-lab.info, but i got no success 
`
at30tse75x init 0
Error initializing AT30TSE75x sensor on I2C #0 @ 0x48
` 
knoten: samr21-10.saclay...

Comment (kfessel) [reply tests/test_interactive.config:1]: of cause this line is wrong

**Comment by kfessel** [comment]
@aabadie: I neither can get the master nor the PR working on the iot-lab.info. I get 
`
at30tse75x init 0
Error initializing AT30TSE75x sensor on I2C #0 @ 0x48
`
in both cases

**Comment by kfessel** [comment]
@aabadie: test worked the senor is at 0x4f ->
```
main(): This is RIOT! (Version: 2021.10-devel-1178-g1ebd4-p-ztimer-at30tse75x)
AT30TSE75x device driver test
> at30tse75x init 0 0x4f
at30tse75x init 0 0x4f
Initialized AT30TSE75x sensor on I2C #0 @ 0x4f
> at30tse75x read
at30tse75x read
Temperature: 20.500 °C
```
maybe you are able to change the link to the in the iot-lab documentation to the  Microchip IO1 Xplained extension doc mentation is wrong maybe [microchips own docu](https://ww1.microchip.com/downloads/en/DeviceDoc/Atmel-42078-IO1-Xplained-Pro_User-Guide.pdf) is the right choice

**Comment by aabadie** [comment]
You forgot to adapt the module dependencies in Kconfig

File: drivers/at30tse75x/Kconfig [hunk 2]
```diff
@@ -10,6 +10,6 @@ config MODULE_AT30TSE75X
     depends on HAS_PERIPH_I2C
     depends on TEST_KCONFIG
     select MODULE_PERIPH_I2C
-    select MODULE_XTIMER
+    select MODULE_ZTIMER_USEC
```
Comment (fjmolinas) [inline drivers/at30tse75x/Kconfig:13]:
```suggestion
    select MODULE_ZTIMER
    select MODULE_ZTIMER_USEC
```

**Comment by fjmolinas** [comment]
> @aabadie: test worked the senor is at 0x4f (riot defaults to 0x48)->
> 
> ```
> main(): This is RIOT! (Version: 2021.10-devel-1178-g1ebd4-p-ztimer-at30tse75x)
> AT30TSE75x device driver test
> > at30tse75x init 0 0x4f
> at30tse75x init 0 0x4f
> Initialized AT30TSE75x sensor on I2C #0 @ 0x4f
> > at30tse75x read
> at30tse75x read
> Temperature: 20.500 °C
> ```
> 
> maybe you are able to change the link to the in the iot-lab documentation to the Microchip IO1 Xplained extension doc mentation is wrong maybe [microchips own docu](https://ww1.microchip.com/downloads/en/DeviceDoc/Atmel-42078-IO1-Xplained-Pro_User-Guide.pdf) is the right choice

Is

https://github.com/RIOT-OS/RIOT/blob/c739516ac4b8f6772ae71a546da05e1d64a7ec5d/drivers/io1_xplained/include/io1_xplained_params.h#L37-L39

correct then?

**Review by fjmolinas — COMMENTED** [review]
Tested, seems to work fine:
`BOARD=samr21-xpro make -C tests/driver_io1_xplained/ flash term -j`
```
2021-11-05 11:27:50,172 # SD Card CID info:
2021-11-05 11:27:50,173 # MID: 3
2021-11-05 11:27:50,173 # OID: SD
2021-11-05 11:27:50,173 # PNM: DSU04
2021-11-05 11:27:50,174 # PRV: 128
2021-11-05 11:27:50,174 # PSN: 4012115276
2021-11-05 11:27:50,174 # MDT: 252
2021-11-05 11:27:50,175 # CRC: 177
2021-11-05 11:27:50,175 # +----------------------------------------+
2021-11-05 11:27:50,176 # 
2021-11-05 11:27:51,155 # Light level: 0
2021-11-05 11:27:51,159 # +-------------------------------------+
2021-11-05 11:27:56,196 # Temperature [°C]: 23.312
2021-11-05 11:27:56,200 # +-------------------------------------+
2021-11-05 11:27:57,208 # SD Card CID info:
2021-11-05 11:27:57,209 # MID: 3
2021-11-05 11:27:57,209 # OID: SD
2021-11-05 11:27:57,210 # PNM: DSU04
2021-11-05 11:27:57,211 # PRV: 128
2021-11-05 11:27:57,213 # PSN: 4012115276
2021-11-05 11:27:57,214 # MDT: 252
2021-11-05 11:27:57,214 # CRC: 177
2021-11-05 11:27:57,218 # +----------------------------------------+
2021-11-05 11:27:57,219 # 
2021-11-05 11:27:58,226 # Light level: 0
2021-11-05 11:27:58,230 # +-------------------------------------+
2021-11-05 11:28:03,267 # Temperature [°C]: 23.187
2021-11-05 11:28:03,271 # +-------------------------------------+
2021-11-05 11:28:04,279 # SD Card CID info:
2021-11-05 11:28:04,280 # MID: 3
2021-11-05 11:28:04,281 # OID: SD
2021-11-05 11:28:04,282 # PNM: DSU04
2021-11-05 11:28:04,282 # PRV: 128
2021-11-05 11:28:04,284 # PSN: 4012115276
2021-11-05 11:28:04,285 # MDT: 252
2021-11-05 11:28:04,286 # CRC: 177
2021-11-05 11:28:04,290 # +----------------------------------------+
2021-11-05 11:28:04,290 # 
2021-11-05 11:28:05,298 # Light level: 0
2021-11-05 11:28:05,301 # +-------------------------------------+

```

File: drivers/at30tse75x/Kconfig (hunk 2 above)
Comment (fjmolinas) [reply drivers/at30tse75x/Kconfig:13]: Please squash right away when this is addressed!

Comment (fjmolinas) [reply drivers/at30tse75x/Kconfig:13]: And re-trigger ci!

**Comment by kfessel** [comment]
> Is
> 
> https://github.com/RIOT-OS/RIOT/blob/c739516ac4b8f6772ae71a546da05e1d64a7ec5d/drivers/io1_xplained/include/io1_xplained_params.h#L37-L39
> 
> correct then?

Yes, I just wasn't using saul which lead me to search ....

File: drivers/at30tse75x/Kconfig (hunk 2 above)
Comment (kfessel) [reply drivers/at30tse75x/Kconfig:13]: doesn't select MODULE_ZTIMER_USEC depend on MODULE_ZTIMER

**Review by fjmolinas — APPROVED** [review]
ACK!

**Comment by kfessel** [comment]
https://ci.riot-os.org/RIOT-OS/RIOT/17137/314bc0090278e175705e06910d82f3e40a6d5430/output/compile/tests/driver_io1_xplained/samr21-xpro:gnu.txt

[moddif.txt](https://github.com/RIOT-OS/RIOT/files/7487132/moddif.txt)
[samr21-xpro gnu.txt](https://github.com/RIOT-OS/RIOT/files/7487133/samr21-xpro.gnu.txt)

Seems like this rule isn't in KConfig: https://github.com/RIOT-OS/RIOT/blob/314bc0090278e175705e06910d82f3e40a6d5430/sys/ztimer/Makefile.dep#L26

**Comment by kfessel** [comment]
I am not sure i this is the way i am supposed to solve that (I would think there is some kind of Automatisation to try to get a working configuration for xtimer but it seem like "No")

**Comment by fjmolinas** [comment]
> I am not sure i this is the way i am supposed to solve that (I would think there is some kind of Automatisation to try to get a working configuration for xtimer but it seem like "No")

@leandrolanzieri is this one of the corner cases where we match configuration to match in the test application or is it something to be fixed in the modeling?

**Comment by kfessel** [comment]
Every kind of trying to auto select the MODULE_XTIMER_ON_ZTIMER failed due to being a loop; Making the choice (of x_on_z or xtimer_compat) non optional would make it allway include one (even if no one need a xtimer interface)

**Comment by leandrolanzieri** [comment]
> @leandrolanzieri is this one of the corner cases where we match configuration to match in the test application or is it something to be fixed in the modeling?

This should be changed in Kconfig modelling. I'm looking into it to provide a fix.

> Making the choice (of x_on_z or xtimer_compat) non optional would make it allway include one (even if no one need a xtimer interface)

Yes, we need to make it non optional but also depend on xtimer being there.

**Comment by aabadie** [comment]
Retested on IoT-LAB and works. ACK
